plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/WorkspaceAwareCanLoadFromDescriptionHelper.java

```diff
@@ -28,8 +29,16 @@
 	@Inject
 	private XWorkspaceManager workspaceManager;
 
+	@Inject
+	private OpenFilesManager openFileManager;
+
 	@Override
 	public Resource createResource(ResourceSet resourceSet, URI resourceURI) {
+		boolean inOpenFileContext = openFileManager.isOpenFileResourceSet(resourceSet);
+		if (inOpenFileContext) {
```

Comment:
Should be fine, because it is basically single-threaded code, as long as resource sets are not passed around across contexts (and then all hell breaks loose, anyway ;). However, these two lines can be made more explicit by just saying `boolean inOpenFileContext = openFilesManager.currentContext() != null;`.